packages/i18n/I18nDecorator/useI18n.js

```diff
@@ -0,0 +1,64 @@
+import useClass from '@enact/core/useClass';
+import React from 'react';
+
+import I18n from './I18n';
+
+/**
+ * Configuration for `useI18n`
+ *
+ * @typedef {Object} useI18nConfig
+ * @memberof i18n/I18nDecorator
+ * @property {String[]}   [latinLanguageOverrides]    Locales that should be treated as latin
+ * @property {String[]}   [nonLatinLanguageOverrides] Locales that should be treated as non-latin
+ * @property {Function}   [onLoadResources]           Called when resources have been loaded after a
+ *                                                    locale change
+ * @property {Function[]} [resources]                 Additional resource callbacks to be invoked
+ *                                                    during a locale change
+ * @property {Boolean}    [sync = false]              Load the resources synchronously
+ * @private
+ */
+
+/**
+ * Object returned by `useI18n`
+ *
+ * @typedef {Object} useI18nInterface
+ * @property {String}  className CSS classes that should be applied to the root node
+ * @property {Boolean} loaded    Indicates if resources have been loaded
+ * @property {String}  locale    Current locale
+ * @property {Boolean} rtl       Indicates the current locale uses right-to-left text direction
+ * @private
+ */
+
+/**
+ * Adds internationalization support
+ *
+ * @param {useI18nConfig} config Configuration options
+ * @returns {useI18nInterface}
+ * @private
+ */
+function useI18n ({locale, ...config} = {}) {
+	const [state, setState] = React.useState({
+		locale,
+		loaded: Boolean(config.sync)
+	});
+
+	const i18n = useClass(I18n, config);
+	i18n.setContext(locale, setState);
+
+	// Add/remove listeners on mount/unmount
+	React.useEffect(() => {
+		i18n.load();
+
+		return () => i18n.unload();
+	}, [i18n]);
```

Comment:
Hello @ryanjduffy 

Many components based on React.Component have `componentDidMount` and `componentWillUnmount` lifecycle methods. So what if providing the lifecycle for hook in the `useClass` like the following way. Then we might remove those redundant hook lifecycle definition in the components and we could have the same method naming like the `load` and the `unload` or something else.
```
function useClass (Ctor, ...args) {
	const ref = React.useRef(null);
	ref.current = ref.current || new Ctor(...args);

	React.useEffect(() => {
		ref.current.load && ref.current.load();

		return () => ref.current.unload && ref.current.unload();
	}, [i18n]);

	return ref.current;
}
```

Reply:
Thanks for the feedback, @ybsung. I'm hesitant to recreate the React lifecycle in a hook but I agree there is some utility here. Curious what others this.

Reply:
Hmm...  I think I would keep the `useEffect` out of `useClass`.  It's easy enough for the consumer to add that functionality if needed and the way args are used it could be tricky to add the `useEffect` body.